service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -131,22 +128,21 @@ private List<DatasetResponse> createDatasetResponses(
         .toList();
   }
 
-  private List<DatasetResponse> collectDatarepoDatasets(AuthenticatedUserRequest user) {
+  private List<DatasetResponse> collectDatasets(
+      AuthenticatedUserRequest user, StorageSystem system) {
     // For this storage system, get the collection of visible datasets and the user's roles for
     // each dataset.
-    var roleMap = datarepoService.getSnapshotInformation(user);
-    return createDatasetResponses(roleMap, StorageSystem.TERRA_DATA_REPO);
-  }
-
-  private List<DatasetResponse> collectWorkspaceDatasets(AuthenticatedUserRequest user) {
-    var roleMap = rawlsService.getWorkspaceInformation(user);
-    return createDatasetResponses(roleMap, StorageSystem.TERRA_WORKSPACE);
+    var roleMap = getService(system).getObjects(user);
+    return createDatasetResponses(roleMap, system);
   }
 
+  // FIXME: possibly refactor to first get all IDs, and then get all datasets. That will help
+  // FIXME: limit the DB work done here.
```

Comment:
I will probably make tech debt tickets for these FIXMEs and remove the comments, rather than fixing the issues now.

Reply:
Done